ui/src/app/dashboard/dashboard.component.ts

```diff
@@ -11,17 +14,47 @@ import {JobStatus} from "../shared/model/JobStatus";
 export class DashboardComponent implements OnInit {
   aggregationResponse: AggregationResponse;
   statusArray: Array<JobStatus> = [JobStatus.Succeeded, JobStatus.Aborted, JobStatus.Running, JobStatus.Failed];
+  timeFrameToDescriptionMap = timeFrameToDescriptionMap;
 
-  constructor(private readonly activatedRoute: ActivatedRoute) {}
+  selectedTimeFrame = defaultTimeFrame;
+  timeFrames: Array<TimeFrame> = [TimeFrame.HOURS1, TimeFrame.HOURS8, TimeFrame.HOURS24,
+    TimeFrame.DAYS7, TimeFrame.DAYS30, TimeFrame.ALLTIME];
+
+  constructor(private readonly router: Router,
+              private readonly activatedRoute: ActivatedRoute) {}
 
   ngOnInit() {
-    this.aggregationResponse = this.activatedRoute.snapshot.data['aggregations'];
+    this.activatedRoute.queryParams.subscribe(() => this.reloadAggregations());
+
+    // make the default toggle option match to url param
+    if (this.activatedRoute.snapshot.queryParams['timeFrame'] != null) {
+      this.selectedTimeFrame = this.activatedRoute.snapshot.queryParams['timeFrame'];
+    }
+  }
 
+  reloadAggregations() {
+    this.aggregationResponse = this.activatedRoute.snapshot.data['aggregations'];
     // collect status
     for (let countEntry of this.aggregationResponse.summary.counts) {
       if (!this.statusArray.includes(countEntry.status)) {
         this.statusArray.push(countEntry.status);
       }
     }
   }
+
+  onTimeFrameChange(newTimeFrame: TimeFrame) {
+    this.selectedTimeFrame = newTimeFrame;
+    const extras = {
```

Comment:
opt_nit: I would probably inline this below instead

Reply:
Fixed.